### DISCUSSION TITLE
QUESTION: Will this library work using USB Host as transport on Arduino Due?

### DISCUSSION THREAD

**Discussion by bptremblay** [FortySevenEffects/arduino_midi_library#212] [Q&A]
I have seen a few examples of MIDI over the Due USB Host, but none for this most excellent library. 
Am I right to think of USB Host mode as "another transport" ? 
Is there work to do to enable this, or is it not possible at this time?

Thanks,
Ben

**Comment by franky47** [comment]
There might be more tricks to handle a Host mode than regular device modes, @lathoub what do you think ?

**Comment by lathoub** [comment]
The USB transport for MIDI is mostly done by [MIDIUSB ](https://github.com/arduino-libraries/MIDIUSB), and the Due is listed as a supported device. I'm unfamiliar with the Due, but see that it has 2 USB's: a native and a programming USB - which do you mean with USB _Host_?

If you get it to work, i'd love to accept an example PR to [USBMIDI](https://github.com/lathoub/Arduino-USBMIDI)

**Reply by bptremblay** [reply]
Meaning specifically I need to use an OTG cable and plug my USB Midi device directly into the Due with no hub and no other device acting as "host".  [DUE]<---USB Host Mode--->[USB Midi Device].
99% of all USB Midi projects assume you want to plug your finished device into a DAW or some other PC software. This is a "DAWless" application, so I want no computer except the Due.

**Reply by bptremblay** [reply]
The only work I have seen toward this goal was from 2015-2018 here> https://github.com/bbx10/USBH_MIDI/tree/due/

But not sure if there was ever success.
I hope this helps.

**Reply by bptremblay** [reply]
If there's no USB Host support, what interface would I need to add this implementation?

**Reply by bptremblay** [reply]
One more thing, USB Host support for MIDI is the "missing link" for making Midi Controllers. When I can buy a control surface like the AKAI APC Mini or the Novation LaunchPads, and simply plug it directly into an Arduino-based project, it enables me to 
1) Use the ready-made controller for my project without needing to make my own User Interface (consider transport controls for drum machine, pads, faders)
2) Use grid controllers like LaunchPad or MIDIPlus Smartpad (or dirt cheap clones) to drive sequencers or pattern generators. 
This also opens up a whole new marketplace for (handmade or fabricated) custom MIDI appliances made from these stock parts.

**Reply by bptremblay** [reply]
(consider this MPC knockoff) https://www.monoprice.com/product?p_id=625894
With USB Host mode on Due MIDI, I could make a "brain" for this device without needing a PC.
